hs_core/hydroshare/utils.py

```diff
@@ -800,8 +800,19 @@ def prepare_resource_default_metadata(resource, metadata, res_title):
     metadata.append({'date': {'type': 'created', 'start_date': resource.created}})
     metadata.append({'date': {'type': 'modified', 'start_date': resource.updated}})
 
-    creator_data = get_party_data_from_user(resource.creator)
-    metadata.append({'creator': creator_data})
+    # only add the resource creator as the creator for metadata if there is not already
+    # creator data in the metadata object
+    creator_data_exists = False
+    for element in metadata:
+        # here k is the name of the element
+        # v is a dict of all element attributes/field names and field values
+        k, v = element.items()[0]
+        if k.lower() == 'creator':
+            creator_data_exists = True
+            break
```

Comment:
@pkdash Wondering whether it is possible to get all keys from metadata, then use ```'creator' in metadata_keys``` to see whether creator metadata element exists. That logic may be simpler.